apps/builder/app/routes/rest.auth.cli.tsx

```diff
@@ -0,0 +1,40 @@
+import { json, redirect, type LoaderArgs } from "@remix-run/node";
+import { sentryException } from "~/shared/sentry";
+
+export const loader = async ({ request }: LoaderArgs) => {
+  try {
+    const currentUrl = new URL(request.url);
+    const returnTo = currentUrl.searchParams.get("returnTo");
+    if (!returnTo) {
+      return {
+        message: "Where are you going?",
+      };
+    }
+    const returnToUrl = new URL(returnTo);
+    const cookies = request.headers.get("Cookie");
+    const _session = cookies
+      ?.split(";")
+      .find((c) => c.trim().startsWith("_session="))
+      ?.split("=")[1];
+    if (!_session) {
+      return redirect(
+        "/login?returnTo=" + currentUrl.pathname + currentUrl.search
+      );
+    }
+    returnToUrl.searchParams.set("token", _session);
```

Comment:
From what I know in auth, you should never expose secrets in url used in browsers. In that case they leak through history, through some android apis etc.
I don't have enough knowledge how cli are usually do logins, but I think that OAUTH device flow is made exactly for this